### DISCUSSION TITLE
Having troubles installing seotoolkit plugin

### DISCUSSION THREAD

**Discussion by eliranrefael** [patrickdemooij9/SeoToolkit.Umbraco#146] [Q&A]
hey how you doing,
i wanted to try out your plugin so i installed the whole package (2.2) to my umbraco (10.3.2) project, simple dotnet run after cleaning the build ofcourse, and i get this 

in this picture we see that although redirect installed, nothing is displayed on dashboard. (same result if you install the whole package)
when you click on redirect it shows nothing, same result if you click on buttons on page level. in packages page the packages seems to be installed (its writing "run pending migartion, when you click on it it says it successfully migrated but the orange button stays)

screens:
![seotools screen1](https://user-images.githubusercontent.com/37138835/211499597-284f49c4-f955-445f-8696-af48a91a1d03.png)
![seotools screen2](https://user-images.githubusercontent.com/37138835/211499601-af369171-36c8-4793-a808-63f939b8f501.jpg)
![seotools screen4](https://user-images.githubusercontent.com/37138835/211499628-c7144d01-777f-401a-9369-356d117a9e2a.jpg)

any ideas of what can cause this strange behaviors?

thanks!
Eliran

**Comment by patrickdemooij9** [comment]
Hi @eliranrefael 
That is very strange.. Not quite sure what would cause this. Also a bit confused on why only the redirects tab is showing even though there should also be other elements there.
Do you see anything in console or your logs?

**Reply by eliranrefael** [reply]
I did this experiment twice, once i download the whole pack and the other i downloaded only one of them, this pictures are from the second experiment, the results are the same so i didnt put pictures from the first experiment aswell.

yes! i get a console error!
![seotoolsConsole](https://user-images.githubusercontent.com/37138835/211509389-a5286363-2d1e-45ff-9472-ffcc8c569314.jpg)

**Reply by patrickdemooij9** [reply]
Seems like an issue with registering the siteaudit controller. This is something that happens when you only download one of the them right? Because the siteaudit shouldn't even load for the redirects package.

Could you try the following?
- Clear your bin folder (I think the SeoToolkit.SiteAudit.dll is still in there even though you uninstalled it). Or do a project clean.
- If there was a SeoToolkit.SiteAudit.dll in your bin folder, then this should fix the above error.
- If the redirects now work, great! Otherwise can you check the console and logs again for a new error?

I'll also try downloading 2.2.0 on a Umbraco 10.3.2 project later today (currently working, so bit limited)

**Reply by eliranrefael** [reply]
thanks for your afforts,
i did what you said, im with the full package now, this is the results:
![seotoolsConsole](https://user-images.githubusercontent.com/37138835/211531810-8f8a6ed0-bc52-4a3b-a17f-a6d5d3cc0f47.jpg)
the clicks is to the dashboard first, then to the first option then to the second.

**Reply by patrickdemooij9** [reply]
Hmmm.. Very strange. I'll give it a go myself to see if I can reproduce it. It might just be something related to Umbraco 10.3.2

**Reply by eliranrefael** [reply]
where are needed assets of the nuget stays in? does it have any assets? i dont see files are created under app_plugin folder, maybe thats related? im running my project on stage and production using the publish command, maybe something needed to be added to the csproj? some assets or content?

**Reply by patrickdemooij9** [reply]
Yes, it does have assets but they are located in your nuget folder as it is served through a razor package library. Whenever you publish it somewhere, it should be located under /wwwroot/App_Plugins/SeoToolkit. 
In your local solution, they should be located at C:\Users\[Your user]\.nuget\packages\seotoolkit.umbraco.sitemap\2.2.0\staticwebassets

**Reply by patrickdemooij9** [reply]
Just downloaded a clean installation of Umbraco 10.3.2 and downloaded SeoToolkit 2.2.0 and I can't reproduce it. I also tried publishing it, but that also copied over the files. Do you only have this issue after publishing your project? Or does it also happen when running the project in visual studio?

Another thing that you could try is clearing the Smidge cache. You can do that by removing the cache folder in the Smidge folder. Maybe the new javascript file isn't being generated properly

**Reply by patrickdemooij9** [reply]
Hi @eliranrefael 
Were you able to get it working by deleting the cache folder in the smidge folder?

**Reply by eliranrefael** [reply]
hey @patrickdemooij9 
sorry for not answering, i was sick unfortunately. i managed to make it work by turning Hosting:Debug:true, once its true and working it can be turned back off. deleting the cache folder didn't work unfortunately. here they explain a bit of why it happens and how to avoid it as a package creator prespective, it appears that the bundler bundle it wrong for some reason.
https://our.umbraco.com/forum/using-umbraco-and-getting-started/108316-frontend-error-the-file-umbraco-backoffice-extensions-js-does-not-contain-any-reference-to-the-extension

another question if i may. many of the settings you open for modification on a doc type level are settings that i may want to leave changeable on the node level, is it possible? i saw you mentioned two properties that can be made changeable in the node, change frequency and priority. what about more settings, like hide from sitemap for example?

another question, how can i create a property like priority in the sitemap section on the node level, its build as a toggler that open a bar, how is it possible to be made?
![image](https://user-images.githubusercontent.com/37138835/214379273-4ef19689-ffff-480c-ab39-056221b88912.png)

thanks!

**Reply by patrickdemooij9** [reply]
Interesting, I didn't know about that setting. I'll make sure to change it for the next release. Thank you for letting me know.

Currently there are two types of modifications in the document type: Meta Tags & Sitemap settings. Currently you can only edit the meta tag values in your node. The sitemap settings will be editable in a future release.

As for the button with slider, that is custom. You can't currently use that as a property editor, but you can take a look at the source and make your own property editor out of it.

**Reply by patrickdemooij9** [reply]
The checkbox slider is located in this file https://github.com/patrickdemooij9/SeoToolkit.Umbraco/blob/dev/Umbraco11/src/SeoToolkit.Umbraco.Sitemap/wwwroot/Sitemap/Displays/DocumentType/sitemapSettings.html

On the bottom of the page
